### ISSUE TITLE
Dockstore UI2 Does Not Always Display Tools for CWL workflows

### ISSUE THREAD

**Issue by coverbeck** [ga4gh/dockstore#1136]
## Bug Report

If you navigate to https://dockstore.org/workflows/Epigenomics-Screw%2FScrew%2Fscrew-preprocess and click Tools, you get a message that "Some tools associated with this workflow are missing from Git repo."

For other workflows, e.g., https://dockstore.org/workflows/ratschlab%2Ftools-cwl-workflow-experiments, it does correctly display the tools.

I don't know what the difference between the workflows is that is causing the problem, but the problem code seems to be in CWLHandler.java

```
private String getDockerHint(List<Object> hints, Gson gsonWorkflow, String currentDefault) {
        if (hints != null) {
            String hintsJson = gsonWorkflow.toJson(hints);
            List<Object> hintsList = gsonWorkflow.fromJson(hintsJson, new TypeToken<List<Object>>() {
            }.getType());

            for (Object requirement : hintsList) {
                Object clazz = ((Map)requirement).get("class");
                if ("DockerRequirement".equals(clazz)) {
                    return ((Map)requirement).get("dockerPull").toString();
                }
            }
        }

        return currentDefault;
```

The for loop should instead probably be:
```
...
for (Object requirement : hintsList) {
                Object dockerRequirement = ((Map)requirement).get("DockerRequirement");
                if (dockerRequirement != null) {
                    return ((Map)dockerRequirement).get("dockerPull").toString();
                }
...
```

**Comment by denis-yuen** [comment]
I think the former wasn't refreshed/indexed properly. Its seems to be missing the underlying tools, not sure why since the data structure is similar to the second one

**Comment by agduncan94** [comment]
I tried forking the repo since with the 1.4.0 release it will properly find the tools. However I still get the issue of a blank tools tab. Also the DAG elements show no associated docker, even if it was in the tool file.  I think you are right and it is not properly parsing hints.

The workflow that you said displays the right info in the tools tab uses requirements whereas the other workflow uses hints. This is more proof that hints probably aren't being parsed correctly.

Seems like the DAG/tools need some TLC. See https://github.com/ga4gh/dockstore/issues/1284

**Comment by garyluu** [comment]
fixed in 1.4.4
